lib/withLoggedInUser.js

```diff
@@ -24,22 +24,41 @@ const maybeRefreshAccessToken = async (currentToken, twoFactorAuthenticatorCode)
 
   if (shouldUpdate) {
     // call to API again to exchange for long term token
-    const res = await refreshToken(currentToken, twoFactorAuthenticatorCode);
+    const res = await refreshToken(currentToken);
     const { token, error } = res;
     if (error) {
-      if (error.code === 401) {
-        return error;
-      }
       return null;
     } else if (token) {
-      setLocalStorage(LOCAL_STORAGE_KEYS.ACCESS_TOKEN, token);
-      return token;
+      const newDecodedToken = jwt.decode(token);
```

Comment:
This whole block looks weird as we're doing the same thing in the 2 conditions.

Reply:
Why returning an encoded token in one case and a decoded token in the other?

Reply:
That's a mistake, thanks for catching. Decoding the token to check the scope but you're right, we need to return the encoded token afterwards.